flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ChangelogModeJsonSerializer.java

```diff
@@ -49,7 +50,12 @@ public void serialize(
             throws IOException {
         jsonGenerator.writeStartArray();
         for (RowKind rowKind : changelogMode.getContainedKinds()) {
-            jsonGenerator.writeString(rowKind.name());
+            if (Objects.requireNonNull(rowKind) == RowKind.DELETE
```

Comment:
```suggestion
            if (rowKind == RowKind.DELETE
```
Why should this be null?